library_generation/utilities.sh

```diff
@@ -114,25 +102,28 @@ download_tools() {
   local protobuf_version=$2
   local grpc_version=$3
   local os_architecture=$4
-  download_generator "${gapic_generator_version}"
+  download_generator_artifact "${gapic_generator_version}" "gapic-generator-java-${gapic_generator_version}.jar"
   download_protobuf "${protobuf_version}" "${os_architecture}"
   download_grpc_plugin "${grpc_version}" "${os_architecture}"
   popd
 }
 
-download_generator() {
+download_generator_artifact() {
   local gapic_generator_version=$1
+  local artifact=$2
+  local project=${3:-"gapic-generator-java"}
   if [ ! -f "gapic-generator-java-${gapic_generator_version}.jar" ]; then
-    if [[ "${gapic_generator_version}" == *"-SNAPSHOT" ]]; then
-      # copy a SNAPSHOT version from maven local repository.
-      copy_from "$HOME/.m2/repository/com/google/api/gapic-generator-java/${gapic_generator_version}/gapic-generator-java-${gapic_generator_version}.jar" \
-      "gapic-generator-java-${gapic_generator_version}.jar"
-      return
+    # first, try to fetch the generator locally
+    local local_fetch_successful=$(copy_from "$HOME/.m2/repository/com/google/api/${project}/${gapic_generator_version}/${artifact}" \
+      "${artifact}")
+    if [[ "${local_fetch_successful}" == "false" ]];then 
+      # download gapic-generator-java artifact from Google maven central mirror if not
+      # found locally
+      >&2 echo "${artifact} not found locally. Attempting a download from Maven Central"
```

Comment:
Can we add logs for successful local fetch or Maven central download? Otherwise it may be misleading that this is the only log we see when we are downloading from Maven Central.

Reply:
thanks, I added a couple success messages for both local and remote fetch